app/presenters/report_presenter.rb

```diff
@@ -24,7 +24,10 @@ def category_percentage(category:)
   def priority_percentage(priority:)
     percentage_for(
       number: Float(defects_completed_on_time(priority: priority).count),
-      total: Float(defects_by_priority(priority: priority).count)
+      total: Float(
+        completed_defects(priority: priority).count +
+        defects_still_open_and_overdue(priority: priority).count
+      )
```

Comment:
Why did this need to change?

Reply:
This is a bit of a tricky calculation to figure out. But as I understood it, the previous implemetation calculated the percentage against the total number of `defects_by_priority` (which could include those without an `actual_completion_date`, those that were rejected etc.

I'm now struggling a bit to get my head around it (and I think this is one of the main problems with the current implementation), but I understood this is "what percentage of defects have been completed on time?" Which would mean checking them against the total number of completed defects (whether completed on time or not) and those that are now overdue but still open (and would therefore not be included in the `completed_defects`).

Again, I'd like to double check this understanding with Hackney, as it feels very complicated....

Reply:
Can we pull this sum into a single method (and `.uniq` it), so it self documents what we thing this set represents?

Reply:
Have done this, still not 100% clear on it, but it's a bit easier to read for developers